frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/VcsRootComboBox.java

```diff
@@ -59,7 +56,8 @@ public synchronized void updateRepositories(@MinLen(1) List<GitRepository> repos
   }
 
   @Override
-  public synchronized Option<GitRepository> getSelectedRepository() {
+  @UIEffect
```

Comment:
That's the simple most important line in this PR: `getSelectedRepository` can only be invoked on UI thread. All other changes on this PR are actually mostly for the sake of adapting to this one.